.github/workflows/sqa-tools-builder.yaml

```diff
@@ -0,0 +1,115 @@
+name: Build SQA Tools
+
+on:
+  push:
+# TODO add these lines back       branches:
+#            - main
+#            - "release_*"
+  pull_request:
+      branches:
+          - main
+          - "release_*"
+
+concurrency:
+    group: sqa-tools-builder-${{ github.ref }}
+    cancel-in-progress: true
+
+env:
+  PW_ENVIRONMENT_ROOT: ${{ github.workspace }}/.environment
+
+jobs:    
+  build-sqa-tools:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/project-chip/chip-build-crosscompile:latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: false
+      
+      - name: Install zip utility
+        shell: bash
+        run: |
+          apt-get update
+          apt-get install -y zip
+     
+      #zip Provision folder
+      - name: Zip provisioning tool
+        shell: bash
+        run: |
+          pwd
+          file="provision.zip"
+          ls -al
+          rm -f "${file}"
+          zip -r "${file}" "provision" -x "provision/support/efr32*" -x "provision/support/si917" -x "provision/modules/__pycache__/*"
+          mkdir -p sqa-tools/
+          mv "${file}" sqa-tools/
+          echo "Provisioning tool zipped successfully: sqa-tools/${file}"
+
+      - name: Mark repository as safe for Git
+        shell: bash
+        run: |
+          git config --global --add safe.directory /__w/matter_extension/matter_extension
+
+      - name: Initialize required submodules
+        shell: bash
+        run: |
+          git submodule update --init third_party/matter_sdk
+      
+      - name: Zip OTA scripts
+        shell: bash
+        run: |
+           ota_file="ota-scripts.zip"
+           zip -r "${ota_file}" "third_party/matter_sdk/scripts/tools/silabs" "third_party/matter_sdk/src/app/ota_image_tool.py" "third_party/matter_sdk/src/controller/python/chip/tlv" -x "*.md" "provision/samples/light/3"
+           mv "${ota_file}" sqa-tools/
+           echo "OTA scripts zipped successfully: sqa-tools/${ota_file}"
+      
+      - name: Checkout matter_sdk submodules
+        shell: bash
+        run: |
+          cd third_party/matter_sdk
+          ./scripts/checkout_submodules.py --platform linux
+          cd ../..
```

Comment:
I don't think we need this cd - each command if i am not mistaken is run from the workplace_root each time.

Reply:
Agreed